### DISCUSSION TITLE
Upgrade to 7.4  'value' is optional in type `getInputProps`

### DISCUSSION THREAD

**Discussion by borachoidev** [mantinedev/mantine#5642] [Q&A] [answered]
Hello,

I am currently using version 7.3.2 and am in the process of migrating to 7.4.2.

During the upgrade, I noticed that the value property in the getInputProps type has been changed to optional. This change has led to a type error in my custom component.

Is the only solution to this error to make the value type in my component optional as well? Or is there another way to resolve this issue?

**Comment by rtivital** [comment, marked as answer]
Types were changed to represent the object returned by `form.getInputProps` more correctly in all cases. If you want to explicitly set value, you can do that by referencing it from `form.values`. For example:

```tsx
<TextInput
  {...form.getInputProps('email')}
  value={form.values.email}
/>
```
